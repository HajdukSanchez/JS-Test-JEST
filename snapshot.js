// This object that we return is a representation of a React component or something like that
export const getCharacter = (data) => ({
  id: data.id,
  name: data.name,
  status: data.status,
  species: data.species,
  gender: data.gender,
})
