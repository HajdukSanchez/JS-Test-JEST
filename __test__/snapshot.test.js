import { getCharacter } from '../snapshot'
// We create this object for simulate something that we have for an API request
import rick from '../rick.json' // Fake API data

describe('Snapshot test', () => {
  test('Should Snapshot', () => {
    expect(getCharacter(rick)).toMatchSnapshot()
    // We need to run the test for generate a snapshot
  })

  test('Should Snapshot always fail', () => {
    // This test always has a fail because her value is not constant
    const user = {
      createAt: new Date(),
      id: Math.floor(Math.random() * 20), // Random element 0-19
    }
    /* expect(user).toMatchSnapshot() */
  })

  test('Should Snapshot with exceptions', () => {
    const user = {
      id: Math.floor(Math.random() * 20), // Dynamic value
      name: 'Jozek Hajduk', // Static value
    }
    expect(user).toMatchSnapshot({
      /* We tell that we don't matter the ID Value */
      id: expect.any(Number), // Here be generate the exception
    })
  })
})
