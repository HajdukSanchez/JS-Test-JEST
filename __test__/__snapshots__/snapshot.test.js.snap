// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Should Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Snapshot test Should Snapshot always fail 1`] = `
Object {
  "createAt": 2021-07-12T22:00:55.081Z,
  "id": 3,
}
`;

exports[`Snapshot test Should Snapshot with exceptions 1`] = `
Object {
  "id": Any<Number>,
  "name": "Jozek Hajduk",
}
`;
